File: ml/src/main/scala/ai/h2o/sparkling/ml/params/HasIgnoredCols.scala
package ai.h2o.sparkling.ml.params

import ai.h2o.sparkling.H2OFrame
import ai.h2o.sparkling.ml.algos.H2OAlgoCommonUtils

trait HasIgnoredCols extends H2OAlgoParamsBase {
  private val ignoredCols =
    new NullableStringArrayParam(this, "ignoredCols", "Names of columns to ignore for training.")

  setDefault(ignoredCols -> null)

  def getIgnoredCols(): Array[String] = $(ignoredCols)

  def setIgnoredCols(value: Array[String]): this.type = set(ignoredCols, value)

  private[sparkling] def getIgnoredColsParam(trainingFrame: H2OFrame): Map[String, Any] = {
    Map("ignored_columns" -> getIgnoredCols())
  }

  override private[sparkling] def getSWtoH2OParamNameMap(): Map[String, String] = {
    super.getSWtoH2OParamNameMap() + ("ignoredCols" -> "ignored_columns")
  }
}

File: py-scoring/src/ai/h2o/sparkling/ml/params/HasIgnoredCols.py
from ai.h2o.sparkling.ml.params.H2OTypeConverters import H2OTypeConverters
from pyspark.ml.param import *


class HasIgnoredCols(Params):
    ignoredCols = Param(
        Params._dummy(),
        "ignoredCols",
        "Names of columns to ignore for training.",
        H2OTypeConverters.toNullableListString())

    def getIgnoredCols(self):
        return self.getOrDefault(self.ignoredCols)

    def setIgnoredCols(self, value):
        return self._set(ignoredCols=value)
